src/Components/Header.tsx

```diff
@@ -0,0 +1,77 @@
+import { useEffect } from 'react';
+import classNames from 'classnames';
+import { Todo } from '../types/Todo';
+
+type HeaderProps = {
+  todos: Todo[];
+  title: string;
+  setTitle: (value: string) => void;
+  setError: (value: string | null) => void;
+  onToggleAll: () => void;
+  handleSubmit: (event: React.FormEvent<HTMLFormElement>) => void;
+  loading: boolean;
+  inputRef: React.RefObject<HTMLInputElement>;
+};
+
+export const Header: React.FC<HeaderProps> = ({
+  todos,
+  title,
+  setTitle,
+  setError,
+  handleSubmit,
+  onToggleAll,
+  loading,
+  inputRef,
+}) => {
+  const areAllTodosCompleted =
+    todos.length > 0 && todos.every(todo => todo.completed);
+
+  useEffect(() => {
+    const timer = setTimeout(() => {
+      inputRef.current?.focus();
+    }, 100);
+
+    return () => clearTimeout(timer);
+  }, [inputRef]);
+
+  return (
+    <header className="todoapp__header">
+      <button
+        type="button"
+        className={classNames(
+          `todoapp__toggle-all ${areAllTodosCompleted ? 'active' : ''}`,
```

Comment:
Use the classnames library rules for add classes with conditions, fix it everywhere